Zend/zend_operators.c

```diff
@@ -836,6 +836,17 @@ ZEND_API void ZEND_COLD zend_incompatible_string_to_long_error(const zend_string
 	zend_error(E_DEPRECATED, "Implicit conversion from float-string \"%s\" to int loses precision", ZSTR_VAL(s));
 }
 
+ZEND_API void ZEND_COLD zend_incompatible_scalar_to_bool_error(zval *arg)
+{
+	if (Z_TYPE_P(arg) == IS_DOUBLE) {
+		zend_error_unchecked(E_DEPRECATED, "Implicit conversion from float %.*H to true, only 0 or 1 are allowed", -1, Z_DVAL_P(arg));
+	} else if (Z_TYPE_P(arg) == IS_LONG) {
+		zend_error(E_DEPRECATED, "Implicit conversion from integer %ld to true, only 0 or 1 are allowed", (long)Z_LVAL_P(arg));
```

Comment:
```suggestion
		zend_error(E_DEPRECATED, "Implicit conversion from int %ld to true, only 0 or 1 are allowed", (long)Z_LVAL_P(arg));
```
Please use the canonical type name